R/model.R

```diff
@@ -0,0 +1,175 @@
+#' Logisic regresion model for the standard curve
+#'
+#' @param dilutions Diltuions used to fit the model
+#' @param mfi MFI values used to fit the model
+#' @param npars Number of parameters to use in the model
+#' @param verbose If `TRUE` prints messages, `TRUE` by default
+#' @param log_dilution If `TRUE` the dilutions are transformed using the `log10` function, `TRUE` by default
+#' @param log_mfi If `TRUE` the MFI values are transformed using the `log10` function, `TRUE` by default
+#' @param scale_mfi If `TRUE` the MFI values are scaled to the range \[0, 1\], `TRUE` by default
+#'
+#' @description
+#' This function uses the `nplr` package to fit the model. The model is fitted using the formula:
+#'
+#' \deqn{y = B + \frac{T - B}{(1 + 10^{b \cdot (x_{mid} - x)})^s},}{y = B + (T - B) / (1 + 10^(b * (x_mid - x)))^s,}
+#'
+#' where:
+#' - \eqn{y} is the predicted value, MFI in our case,
+#' - \eqn{x} is the independent variable, dilution in our case,
+#' - \eqn{B} is the bottom plateau - the right horizontal asymptote,
+#' - \eqn{T} is the top plateau - the left horizontal asymptote,
+#' - \eqn{b} is the slope of the curve at the inflection point,
+#' - \eqn{x_{mid}}{x_mid} is the x-coordinate at the inflection point,
+#' - \eqn{s} is the asymmetric coefficient.
+#'
+#' This equation is refereed as the Richards' equation. More information about the model can be found in the `nplr` package documentation.
+#'
+#' @import nplr
+#' @import dplyr
+#'
+SCModel <- R6::R6Class(
```

Comment:
add function printing the model contents and parameters. The method `print` is in my opinion required